bundle-workflow/src/system/working_directory.py

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from contextlib import contextmanager
+
+saved_path = os.getcwd()
```

Comment:
Needs to be saved before `yield` inside the code of `WorkingDirectory`, otherwise your directory `cd` calls don't stack.

Reply:
Added `saved_path` before `yield`